tools-and-tests/tools/src/main/java/org/hiero/block/tools/days/subcommands/GcpLsCommand.java

```diff
@@ -0,0 +1,374 @@
+     */
+    private final MainNetBucket mainNetBucket;
+
+    /**
+     * Typical wiring: new CommandLine(new GcpLsCommand(mainNetBucket)).execute(args);
+     */
+    public GcpLsCommand() {
+        this.mainNetBucket =
+                new MainNetBucket(true, cacheDir.toPath(), minNodeAccountId, maxNodeAccountId, userProject);
+    }
+
+    @Override
+    public void run() {
+        try {
+            if (!skipLocalCompare && downloadedDaysDir == null) {
+                throw new NotFoundException("DownloadedDaysDir has not been set");
+            }
+
+            validateInputs();
+
+            System.out.printf("Sampling %d (day, hour) slots between %s and %s%n", samples, startDate, endDate);
+
+            long daysRange = ChronoUnit.DAYS.between(startDate, endDate) + 1;
+
+            ExecutorService executor = Executors.newFixedThreadPool(parallelism);
+            try {
+                for (int i = 0; i < samples; i++) {
+                    long dayOffset = ThreadLocalRandom.current().nextLong(daysRange);
+                    LocalDate sampleDate = startDate.plusDays(dayOffset);
+                    int hour = ThreadLocalRandom.current().nextInt(24);
+
+                    handleSample(i + 1, sampleDate, hour, executor);
+                }
+            } finally {
+                executor.shutdownNow();
+            }
+
+        } catch (Exception e) {
+            System.err.println("Error during gcp-ls: " + e.getMessage());
+            e.printStackTrace(System.err);
+        }
+    }
+
+    private void handleSample(
+            final int sampleIndex, final LocalDate sampleDate, final int hour, final ExecutorService executor)
+            throws ExecutionException, InterruptedException {
+
+        System.out.printf("%n=== Sample %d: date=%s hour=%02d ===%n", sampleIndex, sampleDate, hour);
+        System.out.printf("%-8s %-10s %-10s %-10s %-8s%n", "Node", "gcpFiles", "localFiles", "diff", "status");
+
+        // submit node tasks in parallel
+        final List<Future<NodeResult>> futures = new ArrayList<>();
+        for (int node = 0; node < nodeCount; node++) {
+            final int nodeId = node;
+            futures.add(executor.submit(() -> checkNodeForHour(sampleDate, hour, nodeId)));
+        }
+
+        // gather results in node order
+        final List<NodeResult> results = new ArrayList<>(futures.size());
+        for (final Future<NodeResult> future : futures) {
+            results.add(future.get());
+        }
+
+        // print summary table
+        for (final NodeResult r : results) {
+            String status = (r.diff == 0 && r.gcpSigCount >= 0) ? "OK" : "MISMATCH";
+            if (r.gcpSigCount < 0) {
+                status = "ERROR";
+            }
+            System.out.printf(
+                    "%-8s %-10d %-10d %-10d %-8s%n", "0.0." + r.node, r.gcpSigCount, r.localSigCount, r.diff, status);
+        }
+
+        printDifferenceView(results);
```

Comment:
Might be worth a reorder `printView(List<NodeResult> results)` which iterates over `results` and then calls `printDifferenceView(NodeResult result)` and `printLsView(NodeResult result)` based on appropriate flags.
This allows for a single loop of results which could be useful is samples are large